### PR TITLE
122 online sermon

### PR DESCRIPTION
**Pull request by tamarayustian**

### CODE REVIEW

File: ui/src/components/sermons/CurrentSermon.js [hunk 1]
```diff
@@ -76,14 +121,18 @@ const CurrentSermon = ({ currentSermon }) => {
...
           >
-            <iframe
-              width="560"
-              height="315"
-              src={`https://www.youtube.com/embed/${videoUrl}`}
-              title="YouTube video player"
-              allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
-              allowFullScreen
-            ></iframe>
+            {isOnline ? (
```
Comment (albertparedandan) [inline ui/src/components/sermons/CurrentSermon.js:124]: if you have time and wanna do it, i think this file (and the other sermon files) need a major refactor. There are lots of repeated code

File: ui/src/components/sermons/SermonDetails.js [hunk 2]
```diff
@@ -5,18 +5,35 @@ import {
...
+  AccordionIcon,
+} from '@chakra-ui/react';
+import { ChevronLeftIcon } from '@chakra-ui/icons';
+import RelatedSermonCard from './RelatedSermonCard';
+import { customAxios as axios } from '../helpers/customAxios';
+import { DateTime } from 'luxon';
+import { DATE_FULL } from 'luxon/src/impl/formats';
+import React, { useState, useEffect } from 'react';
+import { useHistory } from 'react-router';
+import EventsPage from '../events/EventsPage';
```
Comment (albertparedandan) [inline ui/src/components/sermons/SermonDetails.js:36]: unused

**Review by albertparedandan — CHANGES_REQUESTED** [review]

File: ui/src/components/sermons/SermonDetails.js [hunk 3]
```diff
@@ -84,35 +117,288 @@ const SermonDetails = (props) => {
...
-      sermon.sermonVideoUrl.split("/")[
-        sermon.sermonVideoUrl.split("/").length - 1
-      ];
-    setSermonVideoCode(sermonVideoCode);
+    if (sermon.streamLink === '') {
+      let sermonVideoCode =
+        sermon.sermonVideoUrl.split('/')[
+          sermon.sermonVideoUrl.split('/').length - 1
+        ];
+      setSermonVideoCode(`https://www.youtube.com/embed/${sermonVideoCode}`);
```
[the remark this answers is not in the document]
Comment (tamarayustian) [reply ui/src/components/sermons/SermonDetails.js:125]: like in all the other pages too? its also in get-sermons.js and CurrentSermon.js

File: ui/src/components/sermons/SermonCard.js [hunk 4]
```diff
@@ -5,18 +5,41 @@ import {
...
-} from "@chakra-ui/react";
-import { Link } from "react-router-dom";
-import { DateTime } from "luxon";
+  Icon,
+} from '@chakra-ui/react';
+import { Link } from 'react-router-dom';
+import { DateTime } from 'luxon';
+import { useEffect, useState } from 'react';
+
+const CircleIcon = (props) => (
```
[the remark this answers is not in the document]
Comment (tamarayustian) [reply ui/src/components/sermons/SermonCard.js:14]: apply what change to RelatedSermonCard? the live banner? 
and do you mean in RelatedSermonCard.js? but what is this tho i don't really get it

File: ui/src/components/sermons/SermonDetails.js [hunk 5]
```diff
@@ -25,9 +42,22 @@ const SermonDetails = (props) => {
   const [sermonVideoCode, setSermonVideoCode] = useState();
   const [sermonDate, setSermonDate] = useState();
   const [randomSermons, setRandomSermons] = useState([]);
+  const [onlineSermon, setOnlineSermon] = useState(false);
+  const [noteId, setNoteId] = useState(0);
+  const [isOpen, setIsOpen] = useState(false);
   const currId = props.match.params.id;
   const history = useHistory();
 
+  const onOpen = (e) => {
```
[the remark this answers is not in the document]
Comment (tamarayustian) [reply ui/src/components/sermons/SermonDetails.js:51]: extract as in remove the connection issues button, tabs, etc and make it a separate file/component?

File: ui/src/components/sermons/SermonDetails.js [hunk 6]
```diff
@@ -84,35 +117,288 @@ const SermonDetails = (props) => {
...
+                ></iframe>
+              </AspectRatio>
+            </TabPanel>
+            <TabPanel>
+              <AspectRatio
+                mb="5"
+                width="100%"
+                height={{ base: '100vh', md: '50vh' }}
+              >
+                <iframe src="/events"></iframe>
```
[the remark this answers is not in the document]
Comment (tamarayustian) [reply ui/src/components/sermons/SermonDetails.js:268]: ok so maybe not a tab yea, but like a button maybe?

File: ui/src/components/sermons/SermonCard.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (albertparedandan) [reply ui/src/components/sermons/SermonCard.js:14]: Yes, apply the LIVE banner to RelatedSermonCard.js. it's the component being used when in mobile mode.

File: ui/src/components/sermons/SermonDetails.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (albertparedandan) [reply ui/src/components/sermons/SermonDetails.js:125]: No no, just in this file.

File: ui/src/components/sermons/SermonDetails.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (albertparedandan) [reply ui/src/components/sermons/SermonDetails.js:51]: Yeah. I'm guessing modal open/close is what causes re-render. But extracting the whole tab is cleaner n readable

File: ui/src/components/sermons/SermonDetails.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (albertparedandan) [reply ui/src/components/sermons/SermonDetails.js:268]: Yep

File: ui/src/components/sermons/SermonCard.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (tamarayustian) [reply ui/src/components/sermons/SermonCard.js:14]: Screenshot 2022-01-13 at 8 18 15 PM

got this error, don't know how to fix yet

File: ui/src/components/sermons/RelatedSermonCard.js [hunk 7]
```diff
@@ -1,23 +1,32 @@
...
+    'LLLL dd, yyyy'
   );
-  let sermonImage = "";
+  let sermonImage = '';
   if (sermonData.sermonSeries[0].image !== null)
     sermonImage = sermonData.sermonSeries[0].image.sourceUrl;
-  else sermonImage = process.env.PUBLIC_URL + "/images/ripple_black.svg";
+  else sermonImage = process.env.PUBLIC_URL + '/images/ripple_black.svg';
+
+  const [onlineSermon, setOnlineSermon] = useState(false);
```
Comment (albertparedandan) [inline ui/src/components/sermons/RelatedSermonCard.js:15]: line 15-21 is unused. please make it a habit to check for these things before pushing and asking for review. thanks

File: ui/src/components/sermons/SermonCard.js [hunk 8]
```diff
@@ -35,13 +58,33 @@ const SermonCard = ({ sermonData, allSermons }) => {
...
+                  </Box>
+                )}
+              </>
             </AspectRatio>
             <Box position="absolute" left="100%" top="50%">
               <AspectRatio width="20%" ratio={1 / 1}>
                 <Image
                   borderRadius="100%"
-                  src={process.env.PUBLIC_URL + "/images/PlayButton.png"}
+                  src={process.env.PUBLIC_URL + '/images/PlayButton.png'}
```
Comment (albertparedandan) [inline ui/src/components/sermons/SermonCard.js:87]: [NO-ACTION] lol this file man.. i just realised that idk what this does nor does it show up at all in website.. no need to action on this, just commenting haha

File: ui/src/components/sermons/SermonDetails.js [hunk 9]
```diff
@@ -5,18 +5,29 @@ import {
...
   Container,
-} from "@chakra-ui/react";
-import { ChevronLeftIcon } from "@chakra-ui/icons";
-import RelatedSermonCard from "./RelatedSermonCard";
-import { customAxios as axios } from "../helpers/customAxios";
-import { DateTime } from "luxon";
-import { DATE_FULL } from "luxon/src/impl/formats";
-import React, { useState, useEffect } from "react";
-import { useHistory } from "react-router";
+  Tabs,
```
Comment (albertparedandan) [inline ui/src/components/sermons/SermonDetails.js:11]: these are all no longer used..

File: ui/src/components/sermons/SermonDetails.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (albertparedandan) [reply ui/src/components/sermons/SermonDetails.js:106]: hmm, did you end up doing this?

File: ui/src/components/sermons/SermonDetails.js [hunk 10]
```diff
@@ -84,35 +100,66 @@ const SermonDetails = (props) => {
...
+        <Box bgColor="#F1F1F3" p={5} borderRadius={15}>
+          <Text fontWeight="bold" color="#0628A3" fontSize="md">
+            Description:
+          </Text>
+          <Text>{sermon.sermonDesc}</Text>
+        </Box>
+        <OnlinePageTabs
+          noteId={noteId}
+          sermonNotes={sermon.sermonNotes}
+          refreshCallback={refreshSermonNotes}
```
Comment (albertparedandan) [inline ui/src/components/sermons/SermonDetails.js:157]: [NO-ACTION] no need to action on this, but this whole refreshSermonNotes should ideally be handled by OnlinePageTabs itself.
```
<OnlinePageTabs sermonNotes={sermon.sermonNotes} />
```

File: ui/src/components/sermons/OnlinePageTabs.js [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+  AccordionItem,
+  AccordionButton,
+  AccordionPanel,
+  AccordionIcon,
+  AspectRatio,
+} from '@chakra-ui/react';
+import { ChevronLeftIcon, RepeatIcon } from '@chakra-ui/icons';
+import { useState } from 'react';
+
+const OnlinePageTabs = ({ noteId, sermonNotes, refreshCallback }) => {
```
Comment (albertparedandan) [inline ui/src/components/sermons/OnlinePageTabs.js:19]: [NO-ACTION] related to my comment about `OnlinePageTabs.js` handling refresh sermon notes by itself, just needs 
```
const [noteId. setNoteId] = useState(0);
```

**Review by albertparedandan — APPROVED** [review]

File: ui/src/components/sermons/SermonDetails.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (tamarayustian) [reply ui/src/components/sermons/SermonDetails.js:106]: i just changed `sermonVideoUrl` to `sermonUrl`

File: ui/src/components/sermons/SermonCard.js (hunk 8 above)
Comment (tamarayustian) [reply ui/src/components/sermons/SermonCard.js:87]: lmao...

**Pull request merged**